samples/net/sockets/civetweb_websocket_server/src/http_server_handlers.c

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2020 Alexander Kozhinov
+ * Mail: AlexanderKozhinov@yandex.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(http_server_handlers, LOG_LEVEL_DBG);
+
+#include "http_server_handlers.h"
+
+#define TX_CHUNK_SIZE_BYTES	CONFIG_NET_TX_STACK_SIZE
+
+#define URL_MAIN		"/$"
+#define URL_INDEX_CSS		"/index.css"
+#define URL_INDEX_HTML		"/index.html"
+#define URL_FAVICON_ICO		"/favicon.ico"
+#define URL_WS_JS		"/ws.js"
+
+#define HTTP_TEXT_HTML		"text/html"
+#define HTTP_TEXT_CSS		"text/css"
+#define HTTP_TEXT_JS		"text/javascript"
+#define HTTP_EOFL		"\r\n"  /* http end of line */
+
+#define HTTP_CONTEND_ENCODING	"Content-Encoding: "
+#define HTTP_ENCODING_GZ	"gzip"
+
+#define __code_decl	/* static */
+#define __data_decl	/* static */
+
+/* Typedef for one separate http handler */
+typedef struct {
+	const char *url;
+	int (*hnd)(struct mg_connection *conn, void *cbdata);
+	void *cbdata;
+} http_handler_t;
```

Comment:
I'm also OK with this, I find it quite clean, but checkpatch reports this as a new typedef which is forbidden.

I would personnaly keep the typedef but maybe some others have objections to that ?

Reply:
We have usually used typedefs for function pointers only in zephyr, so that is why the checkpatch complains.

Reply:
@jukkar But using new `typedef` in a `*.c` false is safe, untill this `*.c` file is explicitly included somewhere, which is obviously bad programming behavior. Thus the warning for new `typedf` in `*.c` files is useless! Can you point on somebody, who is familiar with checkpath.pl to add this kind of warnings only for `*.h` files?

Reply:
> But using new `typedef` in a `*.c` false is safe

I would disagree. The reason that we try to avoid extra `typedef` is to help to read / review the code. Using typedef hides the actual type of the variable which sometimes is ok, but usually it is not useful or needed.

Reply:
> > But using new `typedef` in a `*.c` false is safe
> 
> I would disagree. The reason that we try to avoid extra `typedef` is to help to read / review the code. Using typedef hides the actual type of the variable which sometimes is ok, but usually it is not useful or needed.

I Luderstand it now: you following simplification paradigm. Although in my case defining new type is simplification itself. Therefore I see it as necessary in my implementation. If you agree I will leave my typedef at place.

Reply:
> I understand it now. Although in my case defining new typedef is simplification itself. Therefore I see it as necessary in my implementation. If you agree I will leave my typedef at place.

I understand your point but still disagree. Having `struct http_handler foo` is more descriptive than `http_handler_t foo` for the casual reader of the code. I would really prefer than normal struct is used here instead of typedef.

Reply:
Ok. I need then update `this_register_handlers` function since it uses an array of `http_handler_t`. Thus the structure itself will be eliminated. The update comes with next push.